llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -0,0 +1,534 @@
+) -> List[genai.RelevantChunk]:
+    if client is None:
+        client = _default_retriever
+    response = client.query_document(
+        genai.QueryDocumentRequest(
+            name=str(EntityName(corpus_id=corpus_id, document_id=document_id)),
+            query=query,
+            metadata_filters=_convertFilter(filter),
+            results_count=k,
+        )
+    )
+    return list(response.relevant_chunks)
+
+
+@dataclass
+class Passage:
+    text: str
+    ids: List[str]
+
+
+@dataclass
+class TextAnswer:
+    answer: str
+    attributed_passages: List[Passage]
+    answerable_probability: float
+
+
+def generate_text_answer(
+    *,
+    prompt: str,
+    passages: List[str],
+    answer_style: genai.AnswerStyle,
+    client: Optional[genai.TextServiceClient] = None,
+) -> TextAnswer:
+    if client is None:
+        client = _default_text_service
+    response = client.generate_text_answer(
+        genai.GenerateTextAnswerRequest(
+            question=genai.TextPrompt(text=prompt),
+            model=_default_text_service_model,
+            answer_style=answer_style,
+            grounding_source=genai.GroundingSource(
+                passages=genai.InlinePassages(
+                    passages=[
+                        genai.InlinePassage(
+                            text=chunk,
+                            id=str(uuid.uuid4()),
+                        )
+                        for chunk in passages
+                    ]
+                )
+            ),
+        )
+    )
+    return TextAnswer(
+        answer=response.answer.output,
+        attributed_passages=[
+            Passage(text=passage.text, ids=list(passage.passage_ids))
+            for passage in response.attributed_passages
+        ],
+        answerable_probability=0.9,
+    )
+
+
+def _convertToMetadata(metadata: Dict[str, Any]) -> List[genai.CustomMetadata]:
+    cs: List[genai.CustomMetadata] = []
+    for key, value in metadata.items():
+        if isinstance(value, str):
+            c = genai.CustomMetadata(key=key, string_value=value)
+        elif isinstance(value, int):
+            c = genai.CustomMetadata(key=key, int_value=value)
+        elif isinstance(value, float):
+            c = genai.CustomMetadata(key=key, double_value=value)
+        else:
+            raise ValueError(f"Metadata value {value} is not supported")
+
+        cs.append(c)
+    return cs
+
+
+def _convertFilter(fs: Optional[Dict[str, Any]]) -> List[genai.MetadataFilter]:
+    if fs is None:
+        return []
+    assert isinstance(fs, dict)
+
+    filters: List[genai.MetadataFilter] = []
+    for key, value in fs.items():
+        if isinstance(value, str):
+            condition = genai.Condition(
+                operation=genai.Condition.Operator.EQUAL, string_value=value
+            )
+        elif isinstance(value, int):
+            condition = genai.Condition(
+                operation=genai.Condition.Operator.EQUAL, int_value=value
```

Comment:
`int_value` and `double_value` have been replaced by `numeric_value` in `MetadataFilter`.